src/auth/interceptors/createRetryInterceptor.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+const createRetryInterceptor = (options = {}) => {
+  const {
+    httpClient = axios.create(),
+    // This default algorithm is a recreation of what is documented here
+    // https://cloud.google.com/storage/docs/exponential-backoff
+    getBackoffMs = (nthRetry, maximumBackoff = 32000) => {
+      const randomSecond = Math.random();
+      const exponentialBackoff = ((2 ** nthRetry) + randomSecond) * 1000;
+      return Math.min(Math.round(exponentialBackoff), maximumBackoff);
+    },
+    // By default only retry outbound request failures (not responses)
+    shouldRetry = error => !error.response && error.config,
```

Comment:
Why isn't this checking `error.request` like in `processAxiosError`, and why do we want `error.config`?

Reply:
- It's easier to test. We don't get error.request in tests from axios-mock-adapter. In production, axios will return the inner instance of XMLHttpRequest.
- error.config is technically all we need to retry the request
- If error.config is undefined there was some other kind of error that we have introduced in our code (likely in an interceptor)

wdyt?

Reply:
fixed